prov/verbs/src/verbs_cm_xrc.c

```diff
@@ -0,0 +1,450 @@
+/*
+ * Copyright (c) 2018 Cray Inc. All rights reserved.
+ *
+ * This software is available to you under a choice of one of two
+ * licenses.  You may choose to be licensed under the terms of the GNU
+ * General Public License (GPL) Version 2, available from the file
+ * COPYING in the main directory of this source tree, or the
+ * BSD license below:
+ *
+ *     Redistribution and use in source and binary forms, with or
+ *     without modification, are permitted provided that the following
+ *     conditions are met:
+ *
+ *      - Redistributions of source code must retain the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer.
+ *
+ *      - Redistributions in binary form must reproduce the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer in the documentation and/or other materials
+ *        provided with the distribution.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ * BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#include "config.h"
+#include "fi_verbs.h"
+
+#ifdef INCLUDE_VERBS_XRC
+void fi_ibv_save_priv_data(struct fi_ibv_ep *ep, const void *data, size_t len)
+{
+	ep->conn_setup->event_len = MIN(sizeof(ep->conn_setup->event_data),
+					len);
+	memcpy(ep->conn_setup->event_data, data, ep->conn_setup->event_len);
```

Comment:
This is a good question. This usage is consistent with the original usage it modified, and the existing practice is that CM data that is two large is silently dropped (for sending or receiving). I suppose this is because the application can retrieve the amount of CM data it has access to.